app/helpers/application_helper.rb

```diff
@@ -111,7 +111,7 @@ def cloud_cover_url(url)
   end
 
   def tag_colors(tag)
-    Rails.cache.fetch("view-helper-#{tag}/tag_colors", expires_in: 5.hours) do
+    RedisRailsCache.fetch("view-helper-#{tag}/tag_colors", expires_in: 5.hours) do
```

Comment:
This method and thus its call to the cache is one that we could be using a little more efficiently in the view. Here you can see we hit the cache twice to fetch the background and color. 
```
<a class="tag" href="/t/<%= tag %>" style="background-color:<%= tag_colors(tag)[:background] %>;color:<%= tag_colors(tag)[:color] %>">#<%= tag %></a>
```
While we are switching this cache source it would be great if we could update this in the view to store the hash in a variable and then use that variable to grab each key we need rather than hitting the external cache twice.

Reply:
Thanks for the review @mstruve , I strongly agree with you about optimizing the function here, but I am relatively new to ruby, could you point me to a snippet of code or reference, so that I can implement the changes :)

Reply:
This is used in two files
app/views/articles/show.html.erb
app/views/moderations/index.html.erb

In both cases we would want to do something like this:
```
<% @article.cached_tag_list_array.each do |tag| %>
  <% tag_hash = tag_colors(tag) %> 
  <a class="tag" href="/t/<%= tag %>" style="background-color:<%= tag_hash[:background] %>;color:<%= tag_hash[:color] %>">#<%= tag %></a>
<% end %>
```

Reply:
Thanks @mstruve  for this :)